src/OrchardCore/OrchardCore/Modules/Extensions/OrchardCoreBuilderExtensions.cs

```diff
@@ -100,6 +103,22 @@ public static OrchardCoreBuilder AddBackgroundService(this OrchardCoreBuilder bu
                 .Configure<IConfiguration>((options, config) => config
                     .Bind("OrchardCore:OrchardCore_BackgroundService", options));
 
+            builder.Configure(app =>
+            {
+                app.Use((context, next) =>
+                {
+                    // In the background only the endpoints middlewares need to be executed.
+                    if (context.Items.ContainsKey("IsBackground"))
```

Comment:
It may not be a bad idea to store `IsBackground` in a constant since it is used in couple places in the project.

Reply:
Yes I thought about it, maybe in a separate PR where we could create more than one constant, for example I'm thinking about the default host `"localhost"` while in the background.